plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -700,6 +705,12 @@ public ColumnHandle getDeleteRowIdColumnHandle(ConnectorSession session, Connect
     public void dropTable(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
         IcebergTableHandle handle = (IcebergTableHandle) tableHandle;
+        org.apache.iceberg.Table table = getIcebergTable(session, handle.getSchemaTableName());
+        if (table.properties().containsKey(OBJECT_STORE_PATH) ||
+                table.properties().containsKey(WRITE_NEW_DATA_LOCATION) ||
+                table.properties().containsKey(WRITE_METADATA_LOCATION)) {
+            throw new TrinoException(NOT_SUPPORTED, "Table " + handle.getSchemaTableName() + " contains Iceberg path override properties and cannot be dropped from Trino");
+        }
```

Comment:
nit: extract changes to `dropTable` into separate (followup) commit.

Reply:
Not sure what you mean by followup commit, I have separated them into 2 commits, please let me know if this is what you want.

Reply:
I meant exactly that. Thanks.